gk/main.c

```diff
@@ -157,17 +154,39 @@ priority_from_delta_time(uint64_t present, uint64_t past)
 	return integer_log_base_2(delta_time);
 }
 
+static int
+get_grantor_id(struct gk_lpm *ltbl, struct ip_flow *flow)
+{
+	int ret;
+	if (flow->proto == ETHER_TYPE_IPv4)
+		ret = lpm_lookup_ipv4(ltbl->lpm, flow->f.v4.dst);
+	else if (likely(flow->proto == ETHER_TYPE_IPv6))
+		ret = lpm_lookup_ipv6(ltbl->lpm6, flow->f.v6.dst);
+	else
+		rte_panic("Unexpected condition at %s: unknown flow type %hu\n",
+			__func__, flow->proto);
+
+	return ret;
```

Comment:
It's a matter of style, but there's a cleaner way to write this function:

```
if (flow->proto == ETHER_TYPE_IPv4)
	return lpm_lookup_ipv4(ltbl->lpm, flow->f.v4.dst);

if (likely(flow->proto == ETHER_TYPE_IPv6))
	return lpm_lookup_ipv6(ltbl->lpm6, flow->f.v6.dst);

rte_panic("Unexpected condition at %s: unknown flow type %hu\n",
	__func__, flow->proto);

return -1; /* Unreachable. */
```

Perhaps, you can even drop the last `return` and write `return rte_panic(...`.